src/cloudWatchLogs/registry/logStreamRegistry.ts

```diff
@@ -35,17 +35,36 @@ export class LogStreamRegistry {
     /**
      * Adds an entry to the registry for the given URI.
      * @param uri Document URI
-     * @param getLogEventsFromUriComponentsFn Override for testing purposes.
+     * @param initialStreamData Initial Data to populate the registry ActiveTab Data.
      */
     public async registerLog(uri: vscode.Uri, initialStreamData: CloudWatchLogsData): Promise<void> {
         // ensure this is a CloudWatchLogs URI; don't need the return value, just need to make sure it doesn't throw.
         parseCloudWatchLogsUri(uri)
         if (!this.hasLog(uri)) {
             this.setLogData(uri, initialStreamData)
+            this.registerLogHandlers(uri)
             await this.updateLog(uri, 'tail')
         }
     }
 
+    public getOnCloseFuncForUri(uri: vscode.Uri): (document: vscode.TextDocument) => void {
+        return (document: vscode.TextDocument) => {
+            if (document.uri.toString() === uri.toString()) {
+                this.clearStreamIdMap(uri)
+            }
+        }
+    }
+
+    private registerLogHandlers(uri: vscode.Uri): void {
+        vscode.workspace.onDidChangeTextDocument((event: vscode.TextDocumentChangeEvent) => {
+            if (event.document.uri.toString() === uri.toString()) {
+                highlightDocument(this, uri)
+            }
```

Comment:
With a condition like the following, only one global handler is needed:

```suggestion
            if (<is uri in cwl registry?>) {
                highlightDocument(this, uri)
            }
```